app/serializers/api/customer_serializer.rb

```diff
@@ -1,5 +1,14 @@
 module Api
   class CustomerSerializer < ActiveModel::Serializer
     attributes :id, :enterprise_id, :name, :code, :email, :allow_charges
+
+    def attributes
+      hash = super
+      if secret = object.gateway_recurring_payment_client_secret
+        hash.merge!(gateway_recurring_payment_client_secret: secret)
+      end
+      hash.merge!(gateway_shop_id: object.gateway_shop_id) if object.gateway_shop_id
+      hash
+    end
```

Comment:
have you tried appending `gateway_recurring_payment_client_secret` and `gateway_shop_id` to `attributes`? As far as I understand, we want to return nil if they are not set, right? wouldn't we get the same result this way?

Reply:
Matt suggested that we should not return them at all if they're not set, which I think is the correct behavior. I had them originally in the `attributes` list but that returns them in the hash even if they're nil.

Reply:
Going to mark this as resolved but feel free to unresolve 👍